csm_web/scheduler/views/section.py

```diff
@@ -488,3 +487,32 @@ def _student_add(self, request, section):
                 f"<Enrollment:Success> User {log_str(student.user)} enrolled in Section {log_str(section)}"
             )
             return Response({"id": student.id}, status=status.HTTP_201_CREATED)
+
+
+    @action(detail=True, methods=["put"])
+    def change_word_of_day(self, request, pk=None):
```

Comment:
Since this function involves changing the data of SectionOccurrence, I think this should go in a new section_occurrence.py file.

Reply:
Now that I'm looking at the new functions more, I think it'd actually be better to put all of the functions in their own file with their own API endpoint, ex. `/api/wordoftheday/<section_occurrence_pk>/submit` that takes care of both mentor/coord and student requests.

This method would just check if the `request.user` is a mentor/coord for the course (the course can be retrieved from the section occurrence), in which case the PUT request should change the word of the day to the specified one. If the `request.user` is a student of the course, then the PUT request should check the submitted word and update attendance for the student as necessary.

If you'd prefer, you can also split this up into two functions, one for the mentor/coord, and one for the student. In either case, I think it'd be best to move them into their own file, and use the simpler `@api_view` decorator rather than a viewset (especially since these are basically standalone functions with no backing model).

Reply:
If that's the case should I just make the new file and then git revert all of the changes to section.py and student.py (just want to make sure I don't mess up the history more)

Reply:
If by revert you mean reverting to a previous commit entirely, I'd advise not doing that. You can revert the changes in the files manually, but it should be in its own commit; you shouldn't be changing the history at all. In the final rebase, you can always squash the commits together to make it cleaner.